### PR TITLE
Add Coralogix Exporter First PR

### PR DESCRIPTION
**Pull request by ofirshmuel**
**Description:** <Describe what has changed.>

This PR adds a new Coralogix exporter, the exporter sends traces to Coralogix.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.
All tests file in "coralogixexporter" folder

**Documentation:** <Describe the documentation added.>
README and configuration file describe how to use this exporter

### CODE REVIEW

File: exporter/coralogixexporter/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Example configuration:
+```yaml
+exporters:
+  coralogix:
+    # The Coralogix logs ingress endpoint
+    # May also be specified with the CORALOGIX_ENDPOINT environment variable
+    endpoint: "https://api.coralogix.com"
+
+    # Your Coralogix private key (sensitive)
+    # May also be specified with the CORALOGIX_PRIVATE_KEY environment variable
```
Comment (mx-psi) [inline exporter/coralogixexporter/README.md:24]: I assume this functionality (and the rest of environment variables) will be added on a future PR?

File: exporter/coralogixexporter/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Coralogix Exporter
+
```
Comment (mx-psi) [inline exporter/coralogixexporter/README.md:2]: You may want to add a comment saying that this component is in development

File: internal/components/components.go [hunk 3]
```diff
@@ -214,6 +215,7 @@ func Components() (component.Factories, error) {
 		awsxrayexporter.NewFactory(),
 		azuremonitorexporter.NewFactory(),
 		carbonexporter.NewFactory(),
+		coralogixexporter.NewFactory(),
```
Comment (mx-psi) [inline internal/components/components.go:218]: Adding the exporter to `internal/components` should be done on the last PR per the contribution guidelines. The main `go.mod` should have no changes either

File: exporter/coralogixexporter/config_test.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	// "go.opentelemetry.io/collector/config"
+	// "go.opentelemetry.io/collector/config/configtest"
+	// "go.opentelemetry.io/collector/exporter/exporterhelper"
+	// "go.opentelemetry.io/collector/model/pdata"
+)
+
+func TestLoadConfig(t *testing.T) {
+	t.Logf("CORALOGIX CONFIG CHECK : Start To Test")
+
+	// factories, _ := componenttest.NopFactories()
```
Comment (mx-psi) [inline exporter/coralogixexporter/config_test.go:32]: This code should either be removed or uncommented. I think this PR is a good place to add tests for configuration loading, and `TestExporter` can be added later

File: exporter/coralogixexporter/factory.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		Endpoint:   "",
+		PrivateKey: "",
+		AppName:    "",
+		SubSystem:  "",
+	}
+}
+
+func createTraceExporter(_ context.Context, set component.ExporterCreateSettings, config config.Exporter) (component.TracesExporter, error) {
+	// cfg := config.(*Config)
+	set.Logger.Info("Coralogix traces exporter configuration started")
```
Comment (mx-psi) [inline exporter/coralogixexporter/factory.go:49]: The Collector already logs when a component is created or initialized, so this log is not necessary

File: exporter/coralogixexporter/factory.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	return nil, nil
```
Comment (mx-psi) [inline exporter/coralogixexporter/factory.go:50]:
```suggestion
	return nil, fmt.Errorf("Coralogix exporter is not implemented yet")
```

File: exporter/coralogixexporter/factory.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	//exporter := NewCoralogixExporter(cfg, set)
```
Comment (mx-psi) [inline exporter/coralogixexporter/factory.go:51]: You should remove this commented code

**Review by mx-psi — CHANGES_REQUESTED** [review]
Thanks for splitting the original PR. I left some comments, @oded-dd can you review too?

File: exporter/coralogixexporter/README.md (hunk 1 above)
Comment (ofirshmuel) [reply exporter/coralogixexporter/README.md:24]: Yes, this functionality will be added on a future PR.

File: go.mod [hunk 8]
```diff
@@ -479,6 +480,8 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azure
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/carbonexporter => ./exporter/carbonexporter
 
+replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter => ./exporter/coralogixexporter
```
Comment (mx-psi) [inline go.mod:483]: go.mod file should not be modified in this PR (that's why CI is failing)

File: exporter/coralogixexporter/go.mod [hunk 9]
```diff
@@ -0,0 +1,38 @@
+module coralogixexporter
+
+go 1.17
+
+require (
+	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/collector v0.40.0
```
Comment (mx-psi) [inline exporter/coralogixexporter/go.mod:7]: This was bumped to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/42a06f91316730473b00cee4284acaddd88bd3ff/go.mod#L111 can you update here?

**Review by mx-psi — COMMENTED** [review]
Looking good, just a couple more things from my side. Also @oded-dd can you review?

File: exporter/coralogixexporter/go.mod (hunk 9 above)
Comment (ofirshmuel) [reply exporter/coralogixexporter/go.mod:7]: Yes, I will update the changes

**Review by mx-psi — APPROVED** [review]
Looks good from my side if tests pass, I will mark as `ready-to-merge` once we have a review from the other CODEOWNER

**Review by oded-dd — APPROVED** [review]

**Review by mx-psi — COMMENTED** [review]
Looks like there is one more thing to be taken care of (hopefully the last!)

**Comment by jpkrohling** [comment]
Given this week's SIG meeting discussion about new components, should we require a sponsor for this?

**Comment by bogdandrutu** [comment]
@jpkrohling what is Coralogix? Sorry for my ignorance ...

**Comment by oded-dd** [comment]
@bogdandrutu Coralogix is a real-time insights and trend analysis for logs, metrics, tracing (soon once approved :)) and security data with no reliance on storage or indexing

**Comment by oded-dd** [comment]
@mx-psi any update on this PR? Can it be merged?

**Comment by mx-psi** [comment]
@oded-dd @ofirshmuel It looks like part of the lint was not executing before and is now failing because you are missing the [last point in this list](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components) about adding the component to `versions.yaml`.

There is also the sponsoring question raised by @bogdandrutu and @jpkrohling above (but that's a matter to solve among the Collector contrib approvers & maintainers, so it won't block approval)

**Comment by ofirshmuel** [comment]
@mx-psi thanks, I fixed version.yaml, can it be merged?

**Review by mx-psi — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]
We haven't defined how the sponsoring process looks like in its entirety yet. What we agreed was that there should be a sponsor, ideally a volunteer one. If none can be found, we'll round-robin among approvers.

I'm OK with merging this without sorting out the sponsor, as we did define that we will always accept vendor-specific components.

**Pull request merged**